### ISSUE TITLE
When updating object images, old thumbnails are not deleted and updated

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#31809]
### Steps to reproduce
1. Use 8.1 and enable new image system. Set up jpg and webp formats.
2. Visit manufacturer page in FO so that all thumbs are generated.
3. Check `/img/m/` folder and see that manufacturer X has images in all formats.
4. Now go to BO and change the image of this manufacturer.
5. Check `/img/m/` again and see that only JPG thumbnails were updated.

### Problem
- The problem is that we generate only JPG files in BO after uploading them, for performance purposes mostly.
- The situation is the same for manufacturers, categories, stores, suppliers - all objects where the ID of image stay the same.
- It's not a problem for product images because their ID changes and I think we generate all formats directly.

### Temporary solution
- Proper thumbnails are generated after manually regenerating them in `Image settings` in backoffice and clicking "Delete all previous thumbnails".

### Proper solution
- Delete all thumbnails of the given objects after uploading an image in backoffice.

Ping @matthieu-rolland @nicosomb
Originally reproduced by @web-burn in https://github.com/PrestaShop/PrestaShop/pull/31310#issuecomment-1468874335

### PrestaShop version(s) where the bug happened

8.1

**Comment by Codencode** [comment]
@Hlavtox, the solution could be to add the instructions in the `ObjectModelCore::deleteImage()` method?
For example:

```

     public function deleteImage($force_delete = false)
    {
        if (!$this->id) {
            return false;
        }

        if ($force_delete || !$this->hasMultishopEntries()) {
            /* Deleting object images and thumbnails (cache) */
            if ($this->image_dir) {
                if (file_exists($this->image_dir . $this->id . '.' . $this->image_format)
                    && !unlink($this->image_dir . $this->id . '.' . $this->image_format)) {
                    return false;
                }
            }
            if (file_exists(_PS_TMP_IMG_DIR_ . $this->def['table'] . '_' . $this->id . '.' . $this->image_format)
                && !unlink(_PS_TMP_IMG_DIR_ . $this->def['table'] . '_' . $this->id . '.' . $this->image_format)) {
                return false;
            }
            if (file_exists(_PS_TMP_IMG_DIR_ . $this->def['table'] . '_mini_' . $this->id . '.' . $this->image_format)
                && !unlink(_PS_TMP_IMG_DIR_ . $this->def['table'] . '_mini_' . $this->id . '.' . $this->image_format)) {
                return false;
            }
            
            // ***** additional code
            /** @var PrestaShop\PrestaShop\Core\Image\ImageFormatConfiguration $imageFormatConfiguration */
            $imageFormatConfiguration = ServiceLocator::get('PrestaShop\PrestaShop\Core\Image\ImageFormatConfiguration');
            $configuredImageFormats = $imageFormatConfiguration->getGenerationFormats();
            //***********************
             
            $types = ImageType::getImagesTypes();
            foreach ($types as $image_type) {
                if (file_exists($this->image_dir . $this->id . '-' . stripslashes($image_type['name']) . '.' . $this->image_format)
                && !unlink($this->image_dir . $this->id . '-' . stripslashes($image_type['name']) . '.' . $this->image_format)) {
                    return false;
                }
                
                // ***** additional code
                foreach ($configuredImageFormats as $imageFormat) {
                    
                    $file = $this->image_dir . $this->id . '-' . stripslashes($image_type['name']) . '.' . $imageFormat;
                    
                    if (file_exists($file)) {
                        unlink($file);
                    }
                    
                    $file = $this->image_dir . $this->id . '-' . stripslashes($image_type['name']) . '2x.' . $imageFormat;
                    
                    if (file_exists($file)) {
                        unlink($file);
                    }                    
                }
                //***********************
            }
        }

        return true;
    }
```

**Comment by florine2623** [comment]
Hello @Hlavtox ,

I have reproduced the issue with PS 8.1.0.
After following your steps, here's what I have, even after regenerating the images.

<img width="591" alt="Screenshot 2023-06-26 at 10 12 23" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/819a33e6-f2cb-4848-8c23-b5fe450407d2">

I will check the PR right away ^^
Thanks!

**Comment by MatShir** [comment]
Closed by #32493
